rust/lance-io/src/object_store.rs

```diff
@@ -103,6 +103,21 @@ impl std::fmt::Display for ObjectStore {
     }
 }
 
+pub trait ObjectStoreProvider: std::fmt::Debug + Sync + Send {
+    fn new_store(&self, base_path: Url) -> Result<ObjectStore>;
+}
+
+#[derive(Default, Debug)]
+pub struct ObjectStoreRegistry {
+    providers: HashMap<String, Box<dyn ObjectStoreProvider>>,
```

Comment:
I wonder if this should be `Arc<dyn ObjectStoreProvider>>`, since we might want to insert several schemes that point to the same object store. For example, I think Azure has a bunch of different scheme variations.

Reply:
done 👍